### PR TITLE
Add goal centric filter groups in design picker

### PR DESCRIPTION
**Pull request by madhusudhand**
Related to https://github.com/Automattic/dotcom-forge/issues/10014

## Proposed Changes

* Adds filter groups `Features` and `Subjects` to the design picker
* It fallbacks to old view when `multi-selection is false`

image
image

### Note:

Buttons aren't given any background as suggested in the mocks. It will be iterated in a followup PR.
image

## Why are these changes being made?

* To enhance design filtering experience

## Testing Instructions

* Disable the feature `design-picker/goal-centric`
* It should show default view of filters with single selection.
* Now enable `design-picker/goal-centric`
* It should now show two filter groups `Features` and `Subjects` and user should be able to select unselect any of the categories.
* Verify the UI in the small screens.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: packages/design-picker/src/components/unified-design-picker.tsx [hunk 1]
```diff
@@ -207,34 +227,72 @@ const DesignPicker: React.FC< DesignPickerProps > = ( {
 } ) => {
 	const hasCategories = !! Object.keys( categorization?.categories || {} ).length;
 	const filteredDesigns = useFilteredDesigns( designs, categorization );
-
-	// Pick design
+	const features = [ 'blog', 'magazine', 'portfolio', 'podcast', 'store' ];
```
Comment (madhusudhand) [inline packages/design-picker/src/components/unified-design-picker.tsx]: There is not `newsletter` in the list of categories. I have used `magazine` instead.

File: packages/design-picker/src/components/unified-design-picker.tsx [hunk 2]
```diff
@@ -207,34 +227,72 @@ const DesignPicker: React.FC< DesignPickerProps > = ( {
...
+	const [ featureCategories, setFeatureCategories ] = useState< Category[] >( [] );
```
Comment (arthur791004) [inline packages/design-picker/src/components/unified-design-picker.tsx]: I think we can simply use `useMemo` to get features and subjects 🙂

File: packages/design-picker/src/components/unified-design-picker.tsx (hunk 1 above)
Comment (arthur791004) [reply packages/design-picker/src/components/unified-design-picker.tsx:230]: Looks like we have to return the newsletter from the endpoint. I remember we didn't want to show the newsletter category before but I forgot the reason...

Comment (arthur791004) [reply packages/design-picker/src/components/unified-design-picker.tsx]: There are only 2 recommended newsletter designs so the `newsletter` category is hidden. We have to add more designs to display it.

Comment (taipeicoder) [reply packages/design-picker/src/components/unified-design-picker.tsx]: Ola is working on adding more themes to `newsletter`. It's fine if we only render 4 features in the meantime!

File: packages/design-picker/src/components/unified-design-picker.tsx (hunk 2 above)
Comment (madhusudhand) [reply packages/design-picker/src/components/unified-design-picker.tsx]: Good suggestion :) Fixed in d1ab4801a69a8fcb0f1eae8438813610a39d807d

**Review by arthur791004 — APPROVED** [review]
👍

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17049293

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @madhusudhand for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/17049293).

**Pull request merged**